src/Http/Routing/src/Builder/DelegateEndpointRouteBuilderExtensions.cs

```diff
@@ -236,6 +187,77 @@ public static class DelegateEndpointRouteBuilderExtensions
             return new DelegateEndpointConventionBuilder(dataSource.AddEndpointBuilder(builder));
         }
 
+        /// <summary>
+        /// Adds a <see cref="RouteEndpoint"/> to the <see cref="IEndpointRouteBuilder"/> that matches HTTP requests
+        /// for the specified HTTP methods and pattern.
+        /// </summary>
+        /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+        /// <param name="pattern">The route pattern.</param>
+        /// <param name="handler">The delegate executed when the endpoint is matched.</param>
+        /// <param name="httpMethods">HTTP methods that the endpoint will match.</param>
+        /// <returns>A <see cref="IEndpointConventionBuilder"/> that can be used to further customize the endpoint.</returns>
+        public static DelegateEndpointConventionBuilder MapMethods(
+           this IEndpointRouteBuilder endpoints,
+           string pattern,
+           IEnumerable<string> httpMethods,
+           Delegate handler)
+        {
+            if (httpMethods is null)
+            {
+                throw new ArgumentNullException(nameof(httpMethods));
+            }
+
+            var disableInferredBody = false;
+            // GET, DELETE, HEAD, CONNECT, TRACE, and OPTIONS normally do not contain bodies
+            if (!httpMethods.Any(method => !(method.Equals(HttpMethods.Get, StringComparison.Ordinal) ||
+                method.Equals(HttpMethods.Delete, StringComparison.Ordinal) ||
+                method.Equals(HttpMethods.Head, StringComparison.Ordinal) ||
+                method.Equals(HttpMethods.Options, StringComparison.Ordinal) ||
+                method.Equals(HttpMethods.Trace, StringComparison.Ordinal) ||
+                method.Equals(HttpMethods.Connect, StringComparison.Ordinal))))
+            {
+                disableInferredBody = true;
+            }
```

Comment:
```suggestion
            static bool ShouldDisableInferredBody(string method)
            {
                 // GET, DELETE, HEAD, CONNECT, TRACE, and OPTIONS normally do not contain bodies
                return method.Equals(HttpMethods.Get, StringComparison.Ordinal) ||
                       method.Equals(HttpMethods.Delete, StringComparison.Ordinal) ||
                       method.Equals(HttpMethods.Head, StringComparison.Ordinal) ||
                       method.Equals(HttpMethods.Options, StringComparison.Ordinal) ||
                       method.Equals(HttpMethods.Trace, StringComparison.Ordinal) ||
                       method.Equals(HttpMethods.Connect, StringComparison.Ordinal)
            }

            var disableInferredBody = false;
            foreach (var method in httpMethods)
            {
                disableInferredBody = ShouldDisableInferredBody(method);
            }
```

Reply:
Should exit early if a non disabled method is found.

Reply:
Are you suggesting a suggestion for my suggestion?